### PR TITLE
Public endpoints

### PR DESCRIPTION
**Pull request by leilaicruz**
I have finished the list of public enpoints from the wsgi.py file and tried them out and works , once it is working for you we can integrate them in main

### CODE REVIEW

File: documentation/docs/openapi_spec_model.yml [hunk 1]
```diff
@@ -7,234 +7,1424 @@ info:
 servers:
   - url: https://cors-anywhere.herokuapp.com/https://data.4tu.nl
     description: Proxy server to bypass CORS
+  # - url: https://data.4tu.nl/
 
 paths:
-  /portal:
+  /v2/account/applications/authorize:
```
Comment (alwil) [inline documentation/docs/openapi_spec_model.yml:13]: This gives me 404. I think also we could skip the authorize (not sure how is it useful)

File: documentation/docs/openapi_spec_model.yml [hunk 2]
```diff
@@ -7,234 +7,1424 @@ info:
...
+        '200':
+          description: Successfully authorized the application.
+        '400':
+          description: Malformed request.
+        '401':
+          description: Authentication failed.
+        '403':
+          description: User lacks necessary permissions.
+
+  /v2/token:
```
Comment (alwil) [inline documentation/docs/openapi_spec_model.yml:46]: This gives me 404. I think also we could skip the token (not sure how is it useful)

File: documentation/docs/openapi_spec_model.yml [hunk 3]
```diff
@@ -7,234 +7,1424 @@ info:
...
+                      description: The title of the dataset.
+                    description:
+                      type: string
+                      description: A brief description of the dataset.
+        '400':
+          description: Malformed request.
+        '500':
+          description: Internal server error.
+
+  /v2/articles/search:
```
Comment (alwil) [inline documentation/docs/openapi_spec_model.yml:235]: This should be a POST method

File: documentation/docs/openapi_spec_model.yml [hunk 4]
```diff
@@ -7,234 +7,1424 @@ info:
...
+        '500':
+          description: Internal server error.
+
+  /v2/articles/{dataset_id}:
+    get:
+      summary: Retrieve Dataset Details
+      description: Retrieves detailed information about a specific dataset, including its authors, files, custom fields, tags, categories, references, and funding details.
+      tags:
+        - Public Articles
+      parameters:
```
Comment (alwil) [inline documentation/docs/openapi_spec_model.yml:385]: I think the only parameter here is `dataset_id`. I don't see other's in docs or `wsgi.py`. I have also tried them and they don't seem to be showing any results

File: documentation/docs/openapi_spec_model.yml [hunk 5]
```diff
@@ -7,234 +7,1424 @@ info:
...
+          description: The unique identifier of the dataset.
+          schema:
+            type: string
+        - name: version
+          in: path
+          required: true
+          description: The version identifier of the dataset.
+          schema:
+            type: string
+        - name: is_published
```
Comment (alwil) [inline documentation/docs/openapi_spec_model.yml:618]: I don't think it's an actual parameter

File: documentation/docs/openapi_spec_model.yml [hunk 6]
```diff
@@ -7,234 +7,1424 @@ info:
...
+        '500':
+          description: Internal server error.
+
+  /v2/articles/{dataset_id}/versions:
+    get:
+      summary: Retrieve Dataset Versions
+      description: Retrieves a list of versions for a specific dataset, with optional filters for published, latest, or under-review versions.
+      tags:
+        - Public Articles
+      parameters:
```
Comment (alwil) [inline documentation/docs/openapi_spec_model.yml:536]: I think the only parameter is `dataset_id`

**Review by alwil — COMMENTED** [review]
Hi Leila,
 there are some changes to be made to the code. I will branch out and make necessary changes.

File: documentation/docs/openapi_spec_model.yml (hunk 1 above)
Comment (leilaicruz) [reply documentation/docs/openapi_spec_model.yml:13]: Yes , I also got the same response as well, I will remove then

**Comment by leilaicruz** [comment]
Hi, @alwil, I implemented some of the changes you proposed in the YAML spec. If you were also working on this we can merge both branches to main after testing them out , thanks :)

**Pull request merged**